### ISSUE TITLE
Implement an API to get the status of an explicit permissions sync for a Azure dev ops code host connection

### ISSUE THREAD

**Issue by indradhanush** [sourcegraph/sourcegraph#47178]
**Blocked by**

- [ ] https://github.com/sourcegraph/sourcegraph/issues/47165
- [x] https://github.com/sourcegraph/sourcegraph/issues/47177
- [ ] #47167 

/cc @ryphil

**Comment by indradhanush** [comment]
Not needed because #47167 is a wontfix.
